packages/headless/src/controllers/product-listing/facet/headless-product-listing-facet-manager.ts

```diff
@@ -10,7 +11,12 @@ import {
 import {ProductListingSection} from '../../../state/state-sections';
 import {ProductListingEngine} from '../../../app/product-listing-engine/product-listing-engine';
 
-export type {FacetManager, FacetManagerState, FacetManagerPayload};
+export type {
+  CoreFacetManager,
```

Comment:
why does it need to export this type?

Reply:
It is because I made the `FacetManager` interface extend `CoreFacetManager`, so the `CoreFacetManager` has to be exported to be available in the API documentation.

If you think it is better, `FacetManager` can define the same interface as `CoreFacetManager` without extending it. This way the `CoreFacetManager` interface could remain private.

Reply:
It's okay that `CoreFacetManager` gets exported. 

We don't really gain much by redefining the same interface again: net net, the users of the library will get an interface/type of a given shape, that they need to deal with and know about, and that we must support in a backward compatible manner.

The shape of the interface is the important thing, not which files it comes from specifically.